src/table/virtualized-table/hooks/use-dynamic-row-height.ts

```diff
@@ -103,10 +104,43 @@ const useDynamicRowHeight = ({
   );
 
   const getRowHeight = useCallback(
-    (index: number) => rowMeta.current.heights[index] ?? estimatedRowHeight,
+    (rowIdx: number) => rowMeta.current.heights[rowIdx] ?? estimatedRowHeight,
     [rowMeta, estimatedRowHeight]
   );
 
+  /**
+   * Some user actions and events can trigger row heights to be invalidated
+   * - A column is re-sized
+   * - New layout (ex. due to property changes or outside selections)
+   * - Change of page (which would load new data)
```

Comment:
If i'm not mistaking, you mean scrolling right? (which would trigger a fetch of new page data in VT)

Reply:
No scrolling is different event handled by `src/table/virtualized-table/hooks/use-items-rendered-handler.ts`.

VT can still require multiple pages if the number of rows exceeds 250k.